aten/src/ATen/native/StridedRandomAccessor.h

```diff
@@ -140,7 +140,7 @@ class ConstStridedRandomAccessor {
   // 2. |other - this| / this.stride is an Integer.
   C10_HOST_DEVICE
   difference_type operator-(const ConstStridedRandomAccessor& other) const {
-    return (ptr - other.ptr) / stride;
+    return (ptr - other.ptr) / std::max(stride, index_t(1));
```

Comment:
This could potentially be expensive. Would it not be better to test the stride is non-zero at construction time? Also, if the stride is zero, is ptr != other.ptr difference even meaningful? It would seem to me that ptr and other.ptr should always be equal?

Reply:
Oh this is a valid concern, I have refined the code to add a check on `self.stride(dim) == 0` beforehand.

Also this `floating point exception` applies not only to Scalar but also to n-d tensor as long as the `dim` has zero stride, I updated the test cases as well.